### ISSUE TITLE
SecretList: Mismatch between openapi-spec and server behaviour

### ISSUE THREAD

**Issue by loewenstein** [kubernetes/kubernetes#39937]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.): No

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): `SecretList`

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.6", GitCommit:"e569a27d02001e343cb68086bc06d47804f62af6", GitTreeState:"clean", BuildDate:"2016-11-12T05:22:15Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: minikube

**What happened**:
When label filter does not match any Secret the API returns 
```
{\"kind\":\"SecretList\",\"apiVersion\":\"v1\",\"metadata\":{\"selfLink\":\"/api/v1/namespaces/default/secrets\",\"resourceVersion\":\"191989\"},\"items\":null}
```
The openapi spec however states that `items` is a required field.
```
"v1.SecretList": {
"description": "SecretList is a list of Secret.",
"required": [
"items"
],
"properties": {
"apiVersion": {
"description": "APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#resources",
"type": "string"
},
"items": {
"description": "Items is a list of secret objects. More info: http://kubernetes.io/docs/user-guide/secrets",
"type": "array",
"items": {
"$ref": "#/definitions/v1.Secret"
}
},
"kind": {
"description": "Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds",
"type": "string"
},
"metadata": {
"description": "Standard list metadata. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds",
"$ref": "#/definitions/v1.ListMeta"
}
}
},
```
**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):
1. Create a secret without any label
2. List secrets using a labelSelector query parameter
3. Verify that returned JSON contains `"items" : null`

**Anything else do we need to know**:

**Comment by liggitt** [comment]
Dupe of https://github.com/kubernetes/kubernetes/issues/39822

Fixed in master, cherry pick open to the 1.5.x branch and will be fixed in the next point release

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/issues/39888 will fix in 1.5.x
